program_intent_engagement/apps/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,139 @@
+"""
+Tests for the program intent API views
+"""
+import json
+import uuid
+
+import ddt
+from django.urls import reverse
+
+from program_intent_engagement.apps.core.models import ProgramIntent
+from test_utils import ProgramIntentsAPITestCase
+
+
+@ddt.ddt
+class ProgramIntentsViewTests(ProgramIntentsAPITestCase):
+    """
+    Tests Program Intents View
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self.url = reverse('api:v1:program_intent-insert')
+
+        self.program_uuid = str(uuid.uuid4())
+        self.data = {
+            'program_uuid': self.program_uuid,
+            'reason': 'ALL_COURSES_COMPLETE',
+            'certainty': 'CERTAIN_YES',
+            'effective_timestamp': '2022-07-01 00:00:00',
+        }
+
+    def post_api(self, user, data):
+        """
+        Helper function to make a POST request to the API
+        """
+        data = json.dumps(data)
+        headers = self.build_jwt_headers(user)
+
+        return self.client.post(self.url, data, **headers, content_type="application/json")
+
+    def get_response(self, user, data, expected_response):
+        """
+        Helper function to get API response
+        """
+        response = self.post_api(user, data)
+        self.assertEqual(response.status_code, expected_response)
+
+        return response
+
+    # def test_auth_failures(self):
```

Comment:
note: This is commented out for now because there is something sTrAnGe happening with the JWT auth. I posted in dev as well but here's the summary: 
I am adding an API endpoint and when testing, specifically the test_auth_failures [here](https://github.com/edx/program-intent-engagement/pull/11/files), it seems like the jwt authentication class isn't working as expected. It's returning 400 instead of 401, which makes me think it's bypassing the JWT authentication. 
Alie tried to reproduce and she noticed this message (I'm getting it as well):
```
INFO     edx_rest_framework_extensions.auth.jwt.middleware:middleware.py:78 ('The view ProgramIntentsView allows Jwt Authentication. The required permission class, NotJwtRestrictedApplication,', ' was automatically added.')
DEBUG    edx_rest_framework_extensions.permissions:permissions.py:53 Permission JwtRestrictedApplication: returns False
```

We're not sure if this is a false flag though but it seemed a little fishy :tropical_fish:. Does anyone know if there's something that's needed to have this set up properly? Any ideas are greatly appreciated! Thank you!

Reply:
we actually probably want to set JwtRestrictedApplication since you can't call this endpoint successfully without a jwt token - so maybe do that first which would prevent some of those error messages and see how it behaves

it reads like it may not be getting the jwt token at all

I agree that we don't need to test library behavior beyond making sure we hooked up the library, though I do sometimes write temporary tests to understand library behavior.

Reply:
So I think this is now resolved (ty @MichaelRoytman for debugging/investigating with me today and ty @ashultz0 for the details!). Per Michael's suggestion, I ended up adding the `IsAuthenticated` permission to handle the anonymous user case. I did look into `JwtRestrictedApplication` and try that but I wasn't sure if that was the appropriate permission class especially since there weren't other examples I could find. It looks like some type of permission class is required to override the `BasePermission` `has_permission` method which always returns `True`. Lemme know what y'all think!